Test suite hangs on Python 3.6

```
2022-11-27T23:43:05.9749334Z py36-sphinx3 create: /home/runner/work/esbonio/esbonio/lib/esbonio/.tox/py36-sphinx3
2022-11-27T23:43:06.3924125Z py36-sphinx3 installdeps: sphinx>=3,<4, jinja2<3.1
2022-11-27T23:43:14.2369115Z py36-sphinx3 inst: /home/runner/work/esbonio/esbonio/lib/esbonio/.tox/.tmp/package/1/esbonio-0.14.3.tar.gz
2022-11-27T23:43:27.6355505Z py36-sphinx3 installed: alabaster==0.7.12,appdirs==1.4.4,attrs==22.1.0,Babel==2.11.0,certifi==2022.9.24,charset-normalizer==2.0.12,coverage==6.2,dataclasses==0.8,docutils==0.16,esbonio @ file:///home/runner/work/esbonio/esbonio/lib/esbonio/.tox/.tmp/package/1/esbonio-0.14.3.tar.gz,idna==3.4,imagesize==1.4.1,importlib-metadata==4.8.3,importlib-resources==5.4.0,iniconfig==1.1.1,Jinja2==3.0.3,lsp-devtools==0.0.3,MarkupSafe==2.0.1,mock==4.0.3,mypy==0.971,mypy-extensions==0.4.3,packaging==21.3,pluggy==1.0.0,py==1.11.0,pydantic==1.8.2,pygls==0.11.3,Pygments==2.13.0,pyparsing==3.0.9,pyspellchecker==0.7.0,pytest==7.0.1,pytest-asyncio==0.16.0,pytest-cov==4.0.0,pytest-lsp==0.1.3,pytest-timeout==2.1.0,pytz==2022.6,requests==2.27.1,snowballstemmer==2.2.0,Sphinx==3.5.4,sphinxcontrib-applehelp==1.0.2,sphinxcontrib-devhelp==1.0.2,sphinxcontrib-htmlhelp==2.0.0,sphinxcontrib-jsmath==1.0.1,sphinxcontrib-qthelp==1.0.3,sphinxcontrib-serializinghtml==1.1.5,tomli==1.2.3,typed-ast==1.5.4,typeguard==2.13.3,types-appdirs==1.4.3.1,types-docutils==0.19.1.1,types-Pygments==2.13.1.1,types-setuptools==65.6.0.1,typing_extensions==4.1.1,urllib3==1.26.13,zipp==3.6.0
2022-11-27T23:43:27.6358032Z py36-sphinx3 run-test-pre: PYTHONHASHSEED='1225118726'
2022-11-27T23:43:27.6370763Z py36-sphinx3 run-test: commands[0] | python ../../scripts/check-sphinx-version.py
2022-11-27T23:43:27.6981017Z py36-sphinx3 run-test: commands[1] | pytest
2022-11-27T23:43:31.0851752Z ============================= test session starts ==============================
2022-11-27T23:43:31.0852886Z platform linux -- Python 3.6.15, pytest-7.0.1, pluggy-1.0.0
2022-11-27T23:43:31.0874752Z cachedir: .tox/py36-sphinx3/.pytest_cache
2022-11-27T23:43:31.0875731Z rootdir: /home/runner/work/esbonio/esbonio/lib/esbonio, configfile: pyproject.toml
2022-11-27T23:43:31.0876473Z plugins: asyncio-0.16.0, typeguard-2.13.3, timeout-2.1.0, lsp-0.1.3, cov-4.0.0
2022-11-27T23:43:31.0876785Z collected 2491 items
2022-11-27T23:43:31.0876935Z 
2022-11-27T23:43:31.2079645Z tests/doctests/example_directive_option_pattern.txt .                    [  0%]
2022-11-27T23:43:31.2101374Z tests/doctests/example_directive_pattern.txt .                           [  0%]
2022-11-27T23:43:31.2126369Z tests/doctests/example_directive_substitution_pattern.txt .              [  0%]
2022-11-27T23:59:48.5918925Z ##[error]The operation was canceled.
2022-11-27T23:59:48.6019461Z Post job cleanup.
2022-11-27T23:59:48.7758786Z [command]/usr/bin/git version
```

I'm assuming one of the recent changes has broken the language server under Python 3.6

--- a/lib/esbonio/esbonio/lsp/__init__.py
+++ b/lib/esbonio/esbonio/lsp/__init__.py
@@ -123,7 +123,14 @@ def create_language_server(
     if "logger" not in kwargs:
         kwargs["logger"] = logger
 
-    server = server_cls(*args, **kwargs, protocol_cls=Patched)
+    try:
+        server = server_cls(*args, **kwargs, protocol_cls=Patched)
+    except TypeError:
+        # Work around older version of pygls on Python 3.6
+        kwargs.pop("name", None)
+        kwargs.pop("version", None)
+
+        server = server_cls(*args, **kwargs, protocol_cls=Patched)
 
     for module in modules:
         _load_module(server, module)

